### DISCUSSION TITLE
Solving trigonometric equations in wxMaxima

### DISCUSSION THREAD

**Discussion by FEA-eng** [wxMaxima-developers/wxmaxima#1527] [Q&A]
Hi,

apparently trigonometric equations are quite problematic for CAS software in general and for wxMaxima specifically. Is there a way to solve these three exemplary simple trigonometric equations in wxMaxima ?

2*cos(x)^2+4*sin(x)^2=3

sin(x)^4=1-cos(x)^4

sin(20°)*cos(x)+cos(20°)*sin(x)=sqrt(3)/2

I tried with solve function but it doesn't work here. I also tried with to_poly_solve, but again to no avail.

Thanks in advance for your help

**Comment by FEA-eng** [comment]
Ok, apparently I get some solutions with to_poly_solve but I'm not sure if they are correct:
- 1 equation: x=(2π %z791 - (π/2))/2 and x=(2π %z825 + (π/2))/2
- 2 equation: x=(π %z1083)/2
- 3 equation: too long to rewrite here
What are these mysterious symbols like %z791 ? Are they some constants ?

**Reply by daute** [reply]
These equations have infinite (periodic) solutions.

First define the equations (I rewrote them, so that the right hand side is 0 always):

```
eq1:2*cos(x)^2+4*sin(x)^2-3=0;
eq2:sin(x)^4-(1-cos(x)^4)=0;
eq3:sin((20/360)*2*%pi)*cos(x)+cos((20/360)*2*%pi)*sin(x)-sqrt(3)/2=0;
```

Then do a
```
plot2d(lhs(eq1),[x,0,10]);
plot2d(lhs(eq2),[x,0,10]);
plot2d(lhs(eq3),[x,0,10]);

```

(`lhs()` means left hand side of the equation) to see the plots, and where they cross the 0).

the `%znnn` from `to_poly_solve()` are arbitrary integers which are needed for the periodic solutions (see https://maxima.sourceforge.io/docs/manual/maxima_singlepage.html#index-to_005fpoly_005fsolve): 


Especially for trigonometric equations, the solver sometimes needs to introduce an arbitrary integer. These arbitrary integers have the form %zXXX, where XXX is an integer.

So one solution is, where `%znnn = 1`, one where `%znnn = 2`, ...)

If you want numeric solutions, you can solve the equations with the Newton method:

`load("mnewton");`

And then find solutions e.g. near 1:

```
(%i52) mnewton(lhs(eq1),x,1);
(%o52)                    [[x = 0.7853981633974484]]
(%i53) mnewton(lhs(eq2),x,1);
(%o53)                     [[x = 1.570796314567821]]
(%i54) mnewton(lhs(eq3),x,1);
(%o54))                    [[x = 0.6981317007977318]]
```

Best regards, Wolfgang
